README.md

```diff
@@ -1,2 +1,72 @@
-# faraday_connection_pool
+# FaradayConnectionPool
+
 FaradayConnectionPool provides a persistent Net::HTTP Faraday adapter.
+Unlike Net::Persistent, connections are pooled across threads and fibres so that they can be reused more frequently.
+
+## Installation
+
+Add this line to your application's Gemfile:
+
+    gem 'faraday_connection_pool'
+
+And then execute:
+
+    $ bundle
+
+Or install it yourself as:
+
+    $ gem install faraday_connection_pool
+
+
+## Usage
+
+1. If necessary `require 'faraday_connection_pool'`
+
+2. Configure FaradayConnectionPool:
+
+```ruby
+FaradayConnectionPool.configure do |config|
+  config.size = 5 #The number of connections to held in the pool. There is a separate pool for each host/port.
+  config.timeout = 0.5 #If no connection is available from the pool within :timeout seconds the adapter will raise a Timeout::Error.
+end
+```
+
+3. Configure your Faraday connections to use the `:net_http_pooled` adapter provided by the gem:
+
+```ruby
+Faraday.new(:url => 'http://klarna.com') do |conn|
+  conn.adapter :net_http_pooled
+end
+```
+
+## Warning - Retries
+
+FaradayConnectionPool will not automatically try and repair broken connections, so you should configure Faraday to retry
+for you:
+
+```ruby
+Faraday.new do |conn|
+  conn.request :retry, max: 2, interval: 0.05,
+                       interval_randomness: 0.5, backoff_factor: 2
+                       exceptions: [CustomException, 'Timeout::Error']
```

Comment:
I think it's likely that they will want to retry on ::Errno::ECONNRESET - you will get those more when using a keep-alive'd connection. (reading faraday code, it might be that a ECONNRESET is presented as '[Error::ConnectionFailed](https://github.com/lostisland/faraday/blob/626c7f1cf752a8f982a9dcb19103a72844e4cfbc/lib/faraday/adapter/net_http.rb)'... a bit tricky since that combines more than just the ECONNRESET